src/state/pools/hooks.ts

```diff
@@ -23,21 +23,24 @@ import {
 import { State, DeserializedPool, VaultKey } from '../types'
 import { transformPool } from './helpers'
 import { fetchFarmsPublicDataAsync, nonArchivedFarms } from '../farms'
+import { useCurrentBlock } from '../block/hooks'
 
 export const useFetchPublicPoolsData = () => {
   const dispatch = useAppDispatch()
+  const currentBlock = useCurrentBlock()
 
   useSlowRefreshEffect(() => {
     const fetchPoolsDataWithFarms = async () => {
       const activeFarms = nonArchivedFarms.filter((farm) => farm.pid !== 0)
       await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
       batch(() => {
-        dispatch(fetchPoolsPublicDataAsync())
+        dispatch(fetchPoolsPublicDataAsync(currentBlock))
         dispatch(fetchPoolsStakingLimitsAsync())
       })
     }
 
     fetchPoolsDataWithFarms()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We have to use to not trigger this effect by block number

Reply:
I had a thought about avoiding this disable line. Haven't test it yet maybe broken :P  
```ts
// useRefreshEffect.ts
type BlockEffectCallback = (blockNumber: number) => ReturnType<EffectCallback>

export function useFastRefreshEffect(effect: BlockEffectCallback, deps?: DependencyList) {
  const { data = 0 } = useSWR<number>([FAST_INTERVAL, 'blockNumber'])
  const depsMemo = useMemo(() => [data, ...(deps || [])], [data, deps])

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(effect.bind(null, data), depsMemo)
}

useFastRefreshEffect((blockNumber) => {}, [])
```